[tpb] Update Rulesets

funkydude opened the pull request:
Updates some torrent sites which should help some people avoid ISP censorship in light of: https://torrentfreak.com/https-renders-uk-pirate-site-blocklist-useless-160223/

funkydude commented:
@jsha The rules checking script is failing. It is not verbose enough to explain which test it thinks is the "1 out of 2" which is important when you have 6 test URLs added.

funkydude commented:
@jsha @J0WI or whoever, either the script doesn't like test URLs lacking the trailing "/" or it doesn't like the use of "\w+". Either way, it needs fixed, as the output is "not enough URLs".

J0WI commented:
Trailing slashes are required, that's correct.
You can also use ```<rule from="^http:" to="https:" />``` to redirect all target hosts, but you should not redirect any `https` domains, except something is _really_ broken.

funkydude commented:
@J0WI thanks for clarifying. The script should be updated to correctly inform us of that. The current output leaves you guessing of what the problem might be. I'm not sure if I should bother making a ticket, if that will actually go anywhere. (severe lack of dev time)

Regarding https it is correctly used in this case to forward people to the new (correct) domain that may be using bookmarks, but are on censorship enforcing ISPs.

Hopefully I'm done here as this pull is large enough as it is.

funkydude commented:
@J0WI any complaints with this?

J0WI commented:
Please add the trailing slashes anyway, it's quite possible that we will test this in future (ping @fuglede :) ).
I'll need some extra time to test all these domains.

funkydude commented:
@J0WI Thanks, fixed.

J0WI commented on src/chrome/content/rules/Demonoid.xml:
```diff
@@ -0,0 +1,15 @@
+	<!-- Old Domain-->
+	<target host="demonoid.ph" />
+	<target host="www.demonoid.ph" />
+	<rule from="^https?://(www\.)?demonoid\.ph/" to="https://%1demonoid.pw/" />
```
% needs to be a $

J0WI commented on src/chrome/content/rules/IPTorrents.com.xml:
```diff
@@ -6,11 +6,12 @@
 	<target host="pic.iptorrents.com" />
 	<target host="static.iptorrents.com" />
```
There are also s.iptorrents.com and on.iptorrents.com

J0WI commented on src/chrome/content/rules/KickassTorrents.xml:
```diff
@@ -1,20 +1,18 @@
-  <rule from="^http://(?:www\.)?k(?:at\.ph|ickass\.to)/"
-	  to="https://kat.cr/" />
+	<target host="kat.ph" />
+	<target host="www.kat.ph" />
```
for that hosts exist some language specific sub domains, but I think that's okay since they seems to be deprecated anyway.

J0WI commented on src/chrome/content/rules/ThePirateBay.xml:
```diff
@@ -5,25 +5,20 @@
 
 <ruleset name="The Pirate Bay">
```
You may be interested in https://thepiratebay-proxylist.org/ :)

J0WI commented on src/chrome/content/rules/TorrentDay.com.xml:
```diff
@@ -0,0 +1,11 @@
+<ruleset name="Torrent Day">
+
+	<target host="torrentday.com" />
+	<target host="www.torrentday.com" />
```
classic.torrentday.com
editor.torrentday.com
s.torrentday.com
secure.torrentday.com
static.torrentday.com

Refused on:
announce.torrentday.com
irc.torrentday.com

J0WI commented on src/chrome/content/rules/Torrentz.to.xml:
```diff
@@ -1,26 +1,10 @@
-	<!--	Direct rewrites:
-				-->
 	<target host="torrentz.to" />
 	<target host="www.torrentz.to" />
```
mismatch/expired on dev.torrentz.to (redirects to (www.)torrentz-pro.net which would work)

J0WI commented on src/chrome/content/rules/Unblocked.xml:
```diff
@@ -0,0 +1,31 @@
+<!--
+	The domain name for this site frequently changes
```
hosts serve preloaded HSTS header, but is not yet in the list. Could you document that for future maintenance?

J0WI commented:
Some of them are good, I've added notes for those that need additional work.
It's easier to merge when each change get its own PR, you can still link them in a comment.

funkydude commented:
@J0WI Thanks for the review

> % needs to be a $

Oooh, something for the script to catch! :)

> for that hosts exist some language specific sub domains, but I think that's okay since they seems to be deprecated anyway.

Well, they aren't actually country specific. Some of these websites continually change their TLD every time it's added to ISP censorship databases, and deprecate the older ones. I added some forwarding, but I guess I was lazy about forwarding those rules.

> You may be interested in https://thepiratebay-proxylist.org/ :)

I think I'll split that out from TPB and sort it in a separate PR.

> hosts serve preloaded HSTS header, but is not yet in the list. Could you document that for future maintenance?

I don't know what feature this is referring to, or what the list is. Can you elaborate? I added your response into the commit.

> It's easier to merge when each change get its own PR, you can still link them in a comment.

Yup, I bloated this unexpectedly. Get rid of it. :)

J0WI commented on src/chrome/content/rules/Unblocked.xml:
```diff
@@ -0,0 +1,33 @@
+<!--
+	Hosts serve preloaded HSTS header, but is not yet in the list.
```
Oops, could you fix my bad grammar? :D

J0WI commented:
>> % needs to be a $

> Oooh, something for the script to catch! :)

Let me ping @fuglede with that. He's more familiar with the tests.

>> You may be interested in https://thepiratebay-proxylist.org/ :)

> I think I'll split that out from TPB and sort it in a separate PR.

Okay, I'm looking forward to get a PR for that in future. :)


Rules lgtm.

funkydude commented:
@J0WI done

J0WI commented:
Thank you!

fuglede commented:
@J0WI: https://github.com/EFForg/https-everywhere/blob/master/utils/relaxng.xml#L75 would be the place to go for setting up any further URL sanity checking like that. By the way, I noticed that even though unblocked.red specifies the <code>preload</code> token, it does not include <code>includeSubdomains</code> so it is not actually going to end up on the preload lists before that is changed.

J0WI commented:
@fuglede Shouldn't I get a 'Could not resolve host: %1demonoid.pw' anyway?

fuglede commented:
You mean with <code>https-everywhere-checker</code>? If so, then yes, I tried it with the faulty rule target above, and it catches it correctly:

	$ ./fetch-test.sh rules/Demonoid.xml 
	INFO Finished comparing http://demonoid.ph/ -> https://%idemonoid.pw/. Rulefile: rules/Demonoid.xml.
	INFO Finished comparing http://www.demonoid.ph/ -> https://%idemonoid.pw/. Rulefile: rules/Demonoid.xml.
	INFO Finished comparing http://demonoid.pw/ -> https://demonoid.pw/. Rulefile: rules/Demonoid.xml.
	INFO Finished comparing http://www.demonoid.pw/ -> https://www.demonoid.pw/. Rulefile: rules/Demonoid.xml.
	INFO Finished comparing http://demonoid.ph/ -> https://%idemonoid.pw/. Rulefile: rules/Demonoid.xml.
	INFO Finished comparing http://www.demonoid.ph/ -> https://%idemonoid.pw/. Rulefile: rules/Demonoid.xml.
	ERROR rules/Demonoid.xml: Fetch error: http://demonoid.ph/ => https://%idemonoid.pw/: (6, 'Could not resolve host: %idemonoid.pw')
	ERROR rules/Demonoid.xml: Fetch error: http://www.demonoid.ph/ => https://%idemonoid.pw/: (6, 'Could not resolve host: %idemonoid.pw')
	ERROR rules/Demonoid.xml: Fetch error: http://demonoid.ph/ => https://%idemonoid.pw/: (6, 'Could not resolve host: %idemonoid.pw')
	ERROR rules/Demonoid.xml: Fetch error: http://www.demonoid.ph/ => https://%idemonoid.pw/: (6, 'Could not resolve host: %idemonoid.pw')
	INFO Finished in 21.69 seconds. Loaded rulesets: 1, URL pairs: 6.

funkydude commented:
@fuglede lol. You are using %i, not %1. The latter I accidental used as it's the replacement method in other languages.

fuglede commented:
$ ./fetch-test.sh rules/Demonoid.xml 
	INFO Finished comparing http://demonoid.ph/ -> https://%1demonoid.pw/. Rulefile: rules/Demonoid.xml.
	INFO Finished comparing http://www.demonoid.ph/ -> https://%1demonoid.pw/. Rulefile: rules/Demonoid.xml.
	INFO Finished comparing http://demonoid.pw/ -> https://demonoid.pw/. Rulefile: rules/Demonoid.xml.
	INFO Finished comparing http://www.demonoid.pw/ -> https://www.demonoid.pw/. Rulefile: rules/Demonoid.xml.
	INFO Finished comparing http://demonoid.ph/ -> https://%1demonoid.pw/. Rulefile: rules/Demonoid.xml.
	INFO Finished comparing http://www.demonoid.ph/ -> https://%1demonoid.pw/. Rulefile: rules/Demonoid.xml.
	ERROR rules/Demonoid.xml: Fetch error: http://demonoid.ph/ => https://%1demonoid.pw/: (6, 'Could not resolve host: %1demonoid.pw')
	ERROR rules/Demonoid.xml: Fetch error: http://www.demonoid.ph/ => https://%1demonoid.pw/: (6, 'Could not resolve host: %1demonoid.pw')
	ERROR rules/Demonoid.xml: Fetch error: http://demonoid.ph/ => https://%1demonoid.pw/: (6, 'Could not resolve host: %1demonoid.pw')
	ERROR rules/Demonoid.xml: Fetch error: http://www.demonoid.ph/ => https://%1demonoid.pw/: (6, 'Could not resolve host: %1demonoid.pw')
	INFO Finished in 22.61 seconds. Loaded rulesets: 1, URL pairs: 6.

funkydude commented:
@fuglede Well the it worked when I used it :s
https://github.com/EFForg/https-everywhere/pull/4211/commits/48615f952b25fa57e73d40dad17a5fb7db6f93c1
https://travis-ci.org/EFForg/https-everywhere/jobs/113964620

fuglede commented:
Yeah, https-everywhere-checker is not run as part of the Travis test.

funkydude commented:
@fuglede I see, I think there was some confusion here. Basically we were wondering if this can be added to the script so that it's caught in future.
